rasa/core/schemas/domain.yml

```diff
@@ -27,6 +27,33 @@ mapping:
   responses:
     type: "map"
     allowempty: True
+    matching-rule: 'any'
+    mapping:
+      regex;(.+):
+        type: "seq"
+        required: True
+        sequence:
+        - type: "map"
+          allowempty: True
+          matching-rule: 'any'
+          mapping:
+            regex;(^text$):
+              type: "str"
+            regex;(^image$):
+              type: "str"
+            regex;(^custom$):
+              type: "map"
+              allowempty: True
+            regex;(^buttons$):
+              type: "seq"
+              sequence:
+              - type: "map"
+                mapping:
+                  title:
+                    type: "str"
+                  payload:
+                    type: "str"
+
```

Comment:
Can we enforce that they need to have at least one of `(custom, text)`?

Reply:
I don't think we can, using the library we use. I've briefly looked at it and didn't find it. Will look again

Reply:
oh actually, this already validates that one of custom, text, image or buttons is present. Is that enough? You can see the tested config file [here](https://github.com/RasaHQ/rasa/pull/6453/files#diff-45293964fd59daf5bde79922380588cdR17)